osu.Game.Rulesets.Osu/Mods/OsuModRandom.cs

```diff
@@ -42,330 +31,33 @@ public void ApplyToBeatmap(IBeatmap beatmap)
             if (!(beatmap is OsuBeatmap osuBeatmap))
                 return;
 
-            var hitObjects = osuBeatmap.HitObjects;
-
             Seed.Value ??= RNG.Next();
 
             rng = new Random((int)Seed.Value);
 
-            var randomObjects = randomiseObjects(hitObjects);
-
-            applyRandomisation(hitObjects, randomObjects);
-        }
+            var positionModifier = new OsuHitObjectPositionModifier(osuBeatmap.HitObjects);
 
-        /// <summary>
-        /// Randomise the position of each hit object and return a list of <see cref="RandomObjectInfo"/>s describing how each hit object should be placed.
-        /// </summary>
-        /// <param name="hitObjects">A list of <see cref="OsuHitObject"/>s to have their positions randomised.</param>
-        /// <returns>A list of <see cref="RandomObjectInfo"/>s describing how each hit object should be placed.</returns>
-        private List<RandomObjectInfo> randomiseObjects(IEnumerable<OsuHitObject> hitObjects)
-        {
-            Debug.Assert(rng != null, $"{nameof(ApplyToBeatmap)} was not called before randomising objects");
-
-            var randomObjects = new List<RandomObjectInfo>();
-            RandomObjectInfo? previous = null;
             float rateOfChangeMultiplier = 0;
 
-            foreach (OsuHitObject hitObject in hitObjects)
+            foreach (var positionInfo in positionModifier.ObjectPositionInfos)
             {
-                var current = new RandomObjectInfo(hitObject);
-                randomObjects.Add(current);
-
                 // rateOfChangeMultiplier only changes every 5 iterations in a combo
                 // to prevent shaky-line-shaped streams
-                if (hitObject.IndexInCurrentCombo % 5 == 0)
+                if (positionInfo.HitObject.IndexInCurrentCombo % 5 == 0)
                     rateOfChangeMultiplier = (float)rng.NextDouble() * 2 - 1;
 
-                if (previous == null)
+                if (positionInfo == positionModifier.ObjectPositionInfos.First())
                 {
-                    current.DistanceFromPrevious = (float)(rng.NextDouble() * OsuPlayfield.BASE_SIZE.X / 2);
-                    current.RelativeAngle = (float)(rng.NextDouble() * 2 * Math.PI - Math.PI);
+                    positionInfo.DistanceFromPrevious = (float)rng.NextDouble() * OsuPlayfield.BASE_SIZE.Y / 2;
+                    positionInfo.RelativeAngle = (float)(rng.NextDouble() * 2 * Math.PI - Math.PI);
                 }
                 else
                 {
-                    current.DistanceFromPrevious = Vector2.Distance(previous.EndPositionOriginal, current.PositionOriginal);
```

Comment:
Where's this `DistanceFromPrevious` assignment gone? Why isn't it anywhere to be found in the new diff, apparently?

Reply:
That statement only calculates the original jump distance of an object, which is already calculated when the `ObjectPositionInfo` is created in the new version.